### ISSUE TITLE
SSH Key generation broken

### ISSUE THREAD

**Issue by leewillis77** [Block8/PHPCI#800]
Commit 3ade895e30f03db75d9a20f0f6865b097a85ba85 breaks SSH key generation for me, running on Ubuntu 14.10. The issue seems to be that --help isn't a valid option. The output produced by ssh-keygen in this case is all sent to STDERROR, rather than STDOUT, and therefore isn't captured by shell_exec. 

Changing the test in canGenerateKeys() as follows works around the issue for me, but it seems like we should really just try and generate a key and check we have one rather than trying to predict whether ssh-keygen will work?

`$keygen = @shell_exec('ssh-keygen --help 2>&1');`

**Comment by ViGo5190** [comment]
+

**Comment by mavimo** [comment]
Confirmed.

Ubuntu version:
```
root@code-manager:~# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.1 LTS
Release:	14.04
Codename:	trusty
```

SSH Keygen
```
root@code-manager:~# ssh-keygen --help
unknown option -- -
usage: ssh-keygen [options]
// ...
```

With result code:
```
root@code-manager:~# echo $?
1
```

**Comment by tvbeek** [comment]
Does which work on a default windows installation? (I have a lot of linux tools on my windows pc so I can't confirm it)
If that is possible I think it is a good solution

**Comment by leewillis77** [comment]
It'd be good to get some people testing the PR just referenced. It WorksForMe[tm]

**Comment by nicolasiugo** [comment]
In my environment the server has configured the suhosin.executor.func.blacklist directive blocking the function `shell_exec` thus causing a white screen when loading `project/add` .


Maybe a check can be added so that a warning is shown?

**Comment by leewillis77** [comment]
Hi @nicolasiugo,

Presumably that's the same before, and after my patch? If so - it's probably best to open that as a new issue.

**Comment by nicolasiugo** [comment]
Hi, @leewillis77 
yes, the issue still happens after the patch. Removing the "@" from the call `@shell_exec` shows the message:
```
PHP Warning:  shell_exec() has been disabled for security reasons 
```

I'll open a new issue for this. Thanks.

**Comment by nicolasiugo** [comment]
I forgot to mention, I was having the issue regarding the --help option used by the function `canGenerateKeys`. 
The patch fixed problem for me. I'm using CentOS 6.4.

**Comment by dancryer** [comment]
The original bug here is fixed, as @leewillis77's patch has been merged into v1.6.

@nicolasiugo - I'm afraid that PHPCI cannot run without access to `shell_exec`. You'll need to enable that function on your server.
